src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -171,6 +170,28 @@ trait MarathonConf
     true
   }
 
+  private[this] def deriveDefaultAcceptedResourceRolesDefaultBehavior: Option[AcceptedResourceRolesDefaultBehavior] = {
+    defaultAcceptedResourceRoles.toOption match {
+      case Some(set) if set == Set(ResourceRole.Unreserved) => Some(AcceptedResourceRolesDefaultBehavior.Unreserved)
+      case Some(set) if set == Set(mesosRole()) => Some(AcceptedResourceRolesDefaultBehavior.Reserved)
+      case _ => Some(AcceptedResourceRolesDefaultBehavior.Any)
+    }
+  }
+
+  def defaultAcceptedResourceRolesSet(serviceRole: Role): Set[String] = {
+    acceptedResourceRolesDefaultBehavior() match {
+      case AcceptedResourceRolesDefaultBehavior.Any => Set(serviceRole, ResourceRole.Unreserved)
+      case AcceptedResourceRolesDefaultBehavior.Unreserved => Set(ResourceRole.Unreserved)
+      case AcceptedResourceRolesDefaultBehavior.Reserved => Set(serviceRole)
+    }
+  }
+
+  lazy val acceptedResourceRolesDefaultBehavior = opt[AcceptedResourceRolesDefaultBehavior](
+    name = "accepted_resource_roles_default_behavior",
+    descr = "Default behavior for acceptedResourceRoles if not explicitly set on a service." +
+      "This defaults to 'any', which allows either unreserved or reserved resources",
+    default = deriveDefaultAcceptedResourceRolesDefaultBehavior)(acceptedResourceRolesDefaultBehaviorConverter)
```

Comment:
Should [this sanitation method in AppNormalization](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/api/v2/AppNormalization.scala#L209) use this value instead of `Unreserved`?

Reply:
Discussed and nope.

Reply:
As discussed elsewhere, I don't think so. We really just need to get to a point where we remove sanitization of acceptedResourceRole values.